src/lib/rnp.c

```diff
@@ -1147,6 +1149,148 @@ rnp_decrypt_file(rnp_ctx_t *ctx, const char *f, const char *out)
     return ret ? RNP_SUCCESS : RNP_ERROR_GENERIC;
 }
 
+typedef struct pgp_parse_handler_param_t {
+    char in[PATH_MAX];
+    char out[PATH_MAX];
+} pgp_parse_handler_param_t;
+
+/* process the pgp stream */
+
+bool
+rnp_parse_handler_dest(pgp_parse_handler_t *handler, pgp_dest_t *dst, const char *filename)
+{
+    pgp_parse_handler_param_t *param = handler->param;
+
+    if (!param) {
+        return false;
+    }
+
+    if (strlen(param->out) > 0) {
+        return init_file_dest(dst, param->out) == RNP_SUCCESS;
+    } else {
+        return init_stdout_dest(dst) == RNP_SUCCESS;
+    }
+}
+
+int
+rnp_process_stream(rnp_ctx_t *ctx, const char *in, const char *out)
+{
+    pgp_source_t               src;
+    rnp_result_t               err;
+    pgp_parse_handler_t *      handler = NULL;
+    pgp_parse_handler_param_t *param = NULL;
+    int                        result;
+
+    if (in && (strlen(in) > sizeof(param->in))) {
```

Comment:
I think that's using `PATH_MAX` make this code easy to read but it also style question

Reply:
After reading some stuff about PATH_MAX i think we should not rely on it. Actually, better would be to use char * instead of pre-allocated arrays.